File: week3-AppDev-crud/crud-nodejs/Ex07_Query4a_ListJourneys.js
const connection = require('./db_connection')

console.log("========================================")
console.log("Start exercise")

const selectJourneys  = 'select * from spacecraft_journey_catalog where spacecraft_name=?'
const spacecraft_name = 'Crew Dragon Endeavour,SpaceX'

connection.client
.execute(selectJourneys, [spacecraft_name])
.then(function(result){
    result.rows.forEach(row => {
        console.log('- Journey: %s Summary: %s', row.journey_id.toString(), row.summary)
    })
    connection.client.shutdown()
    console.log("SUCCESS")
})
.catch(function(error){
    console.log(error.message)
    connection.client.shutdown()
    console.log("FAILED")
});

console.log("========================================")

File: week3-AppDev-crud/crud-python/Ex07_Query4a_ListJourneys.py
#!/usr/bin/env python3
from db_connection import Connection
import uuid

print('========================================')
print('Start exercise')
try:
    connection = Connection()
    output = connection.session.execute(
        "select * from spacecraft_journey_catalog where spacecraft_name=%s",
        ['Crew Dragon Endeavour,SpaceX']
    )
    for row in output:
       print('- Journey;', row.journey_id, ' Summary:', row.summary)
       
except Exception as e: 
    print(e)
    print('Failure')
else:

    print('Success')
    print('Closing connection (up to 10s)')
finally:
    connection.close()
print('========================================')
